### PR TITLE
Replace `UTF8Encoding(false)` with `Encoding.Default`

### PR DESCRIPTION
**Pull request by xtqqczze**
See: dotnet/runtime#45541

### CODE REVIEW

File: src/System.Management.Automation/engine/Utils.cs [hunk 1]
```diff
@@ -1380,9 +1380,6 @@ internal static bool Succeeded(int hresult)
         //     Add-Member ScriptProperty Preamble { $this.GetEncoding().GetPreamble() -join "-" } -PassThru |
         //     Format-Table -Auto
 
-        internal static readonly UTF8Encoding utf8NoBom =
-            new UTF8Encoding(encoderShouldEmitUTF8Identifier: false);
-
```
Comment (iSazonov) [inline src/System.Management.Automation/engine/Utils.cs:1385]: I'd prefer to convert this to const and keep all references for history:
`internal const UTF8Encoding Utf8NoBom = Encoding.Default;`

File: src/System.Management.Automation/engine/NativeCommandProcessor.cs [hunk 2]
```diff
@@ -2054,14 +2054,13 @@ internal void Start(Process process, NativeCommandIOFormat inputFormat)
         {
             Dbg.Assert(process != null, "caller should validate the paramter");
 
-            // Get the encoding for writing to native command. Note we get the Encoding
-            // from the current scope so a script or function can use a different encoding
-            // than global value.
```
Comment (daxian-dbw) [inline src/System.Management.Automation/engine/NativeCommandProcessor.cs:2059]: Please keep this comment

File: src/System.Management.Automation/engine/Utils.cs (hunk 1 above)
Comment (daxian-dbw) [reply src/System.Management.Automation/engine/Utils.cs:1385]: I think using `Ecoding.Default` directly at the call sites is more friendly to someone who is reading the code.

**Review by daxian-dbw — COMMENTED** [review]
LGTM overall with a minor comment. Thanks for making this change @xtqqczze!

**Review by iSazonov — APPROVED** [review]

**Comment by iSazonov** [comment]
@daxian-dbw Can you approve?

**Review by daxian-dbw — APPROVED** [review]
LGTM

**Comment by iSazonov** [comment]
@xtqqczze Thanks!

**Pull request merged**